### PR TITLE
Bug 1272329: innobackupex dies if a directory is not readable. i.e.: …

### PR DESCRIPTION
**Pull request by VasilyNemkov**
…lost+found

Introducing new options: --tables-exclude and --databases-exclude that work
similarly to --tables and --databases, but exclude given names/paths from backup.
Updated a xb_databases_options test to verify new options interoperation with existing ones.

Unreadable directory can now be excluded with --databases-exclude=path, see test for details.

Param-build: http://jenkins.percona.com/view/PXB%202.3/job/percona-xtrabackup-2.3-param/245/

### CODE REVIEW

File: storage/innobase/xtrabackup/src/xtrabackup.cc [hunk 1]
```diff
@@ -2240,43 +2250,136 @@ xtrabackup_io_throttling(void)
...
+    DATABASE_DONT_SKIP,
+    DATABASE_DONT_SKIP_UNLESS_EXPLICITLY_EXCLUDED,
+};
+
+/************************************************************************
+Checks if a database specified by name should be skipped from backup based on
+the --databases, --databases_file or --databases_exclude options.
 
-			return(TRUE);
+@return TRUE if entire database should be skipped,
```
Comment (VasilyNemkov) [inline storage/innobase/xtrabackup/src/xtrabackup.cc]: Ooops, stale comment.

**Comment by gl-sergei** [comment]
Same comments as 2.3

**Comment by VasilyNemkov** [comment]
Fixed

**Comment by gl-sergei** [comment]
OK to merge if jenkins tests are passed

**Comment by VasilyNemkov** [comment]
Param-build: http://jenkins.percona.com/view/PXB%202.3/job/percona-xtrabackup-2.4-param/148/

**Pull request merged**